### PR TITLE
Add script to generate gource

### PR DESCRIPTION
**Pull request by philberty**
This script tries to isolate our changes to GCC minus the changes to
gcc/config so that we can visualize our development history.

I will be uploading a video output to youtube shortly, it also includes our cargo-integration project too :D

### CODE REVIEW

**Review by flip1995 — COMMENTED** [review]
I recommend running shellcheck over this. I did and the output was pretty big.

File: gcc/rust/gource-gccrs.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    *.md \
+    gcc/rust \
+    gcc/testsuite/rust \
+    gcc/testsuite/rust.test > gccrs-raw.log
+cat gccrs-raw.log | gource --log-format git --output-custom-log gccrs.log -
+rm -f gccrs-raw.log
+sed -i -E "s#(.+)\|#\1|/compiler#" cargo.log
+
+# combine them
+combined_log="$(mktemp /tmp/gource.XXXXXX)"
```
Comment (dkm) [inline gcc/rust/gource-gccrs.sh:72]: Maybe remove the temp file after its use or using `trap .... EXIT` ?

**Comment by philberty** [comment]
> I recommend running shellcheck over this. I did and the output was pretty big.

Thanks for the tip I was not aware of shellcheck, I've fixed all but the awk line warning not sure how I can fix it yet.

File: gcc/rust/gource-gccrs.sh (hunk 1 above)
Comment (philberty) [reply gcc/rust/gource-gccrs.sh:90]: Added a cleanup step instead of a trap because sometimes killing to keep the tmp files can be useful.

**Comment by philberty** [comment]
Here is the youtube! https://www.youtube.com/watch?v=QkdGY9VljTc

**Review by CohenArthur — APPROVED** [review]
Very cute present :)

**Pull request merged**